### ISSUE TITLE
Android Auto won't connect with latest AA

### ISSUE THREAD

**Issue by doctordbx** [rsjudka/intelligent-auto#40]
So this seems to be a common problem with most AASDK based implementations. This problem doesn't exist in OpenAuto Pro which I have, but in intelligent-auto the problem is there.

[2020-06-06 20:37:17.752954] [0x00007fb26ed1f700] [info]    [OpenAuto] [App] Waiting for device...
[2020-06-06 20:37:17.752974] [0x00007fb26ed1f700] [error]   [OpenAuto] [AudioInputService] channel error: AaSdk error code: 30, native code: 0
[2020-06-06 20:37:17.752982] [0x00007fb26ed1f700] [error]   [OpenAuto] [AudioService] channel error: AaSdk error code: 30, native code: 0, channel: SPEECH_AUDIO
[2020-06-06 20:37:17.752996] [0x00007fb26ed1f700] [error]   [OpenAuto] [AudioService] channel error: AaSdk error code: 30, native code: 0, channel: SYSTEM_AUDIO
[2020-06-06 20:37:17.753000] [0x00007fb26ed1f700] [error]   [OpenAuto] [SensorService] channel error: AaSdk error code: 30, native code: 0
[2020-06-06 20:37:17.753004] [0x00007fb26ed1f700] [error]   [OpenAuto] [VideoService] channel error: AaSdk error code: 30, native code: 0
[2020-06-06 20:37:17.753008] [0x00007fb26ed1f700] [error]   [OpenAuto] [AndroidAutoEntity] channel error: AaSdk error code: 30, native code: 0
[2020-06-06 20:37:17.753015] [0x00007fb26ed1f700] [debug]   [OpenAuto] [AndroidAutoEntity] destroy.
[2020-06-06 20:37:17.753021] [0x00007fb26f520700] [error]   [OpenAuto] [AudioService] channel error: AaSdk error code: 30, native code: 0, channel: MEDIA_AUDIO
[2020-06-06 20:37:17.753046] [0x00007fb26fd21700] [error]   [OpenAuto] [BluetoothService] channel error: AaSdk error code: 30, native code: 0

I also have the same problem in crankshaft.

**Comment by rhysmorgan134** [comment]
What branch did you make it on?

**Comment by doctordbx** [comment]
Develop branch. This is just running it on Linux though running it in VirtualBox, but I had the same issue with Crankshaft on the Pi... which is why I went to OpenAuto Pro.

**Comment by rhysmorgan134** [comment]
Did you also do a "git pull --recursive <repo>"

This one definately is working, it can just be abit tricky making sure you have all the correct branches on the submodules too.

I'm just in the process of doing a vid for installing it from scratch.

But im sure @rsjudka will be along to help out

**Comment by doctordbx** [comment]
Yep, also did that. Even manually pulled in the latest patch from this fork:

**Comment by rsjudka** [comment]
Just to keep things documented in case someone else runs into this issue, how did you resolve it?

also, the aasdk develop branch actually has ice's fix just an fyi

**Comment by rhysmorgan134** [comment]
@doctordbx give this a go

https://youtu.be/ElS7Vx9JuL0

**Comment by doctordbx** [comment]
I didn't solve it really but have yet to try it out on the pi. It still has
intermittent issues connected to the desktop.

**Comment by rsjudka** [comment]
Do the intermittent issues occur only when playing some sort of audio? I wonder if ice's fix wasn't enough to deal with all possible issues here.

What OS are you using? Have you tried different USB cables, ports? Have you tried running wireless AA?

**Comment by doctordbx** [comment]
Generally the issues crop up when connecting the phone via USB or headless,
or switching between the two. Maybe 1 out of every 4 of 5 times it won't
connect and instead gives the error 30.

I have the same problem with Crankshaft direct to Pi (worse actually, I'd
say 4/5 don't work with Crankshaft ) but no issue with Open Auto Pro.

OS is Mint Linux 19 - both in virtual box on Mac and onto a different
MacBook running Linux natively
I have tried different cables and USB ports
I have not tried on the Pi yet.

Other than great work though. I'm poking through the rest of it looking at
ways I can interface the CAN bus directly into it (which I have already
accomplished).

**Comment by rsjudka** [comment]
hmm interesting i have not experienced this issue, but then again i haven't really done much testing in terms of the openauto connection. I'll try to see if I can reproduce it and see where its going wrong.

You got it working in a VM? I know someone around here tried doing that before but was having some issues with the USB connection. Did you need to do anything special to set that up?

In what ways are you interfacing with the CAN bus? I would be curious to see, maybe get some inspiration from it :smiley:

**Comment by doctordbx** [comment]
I was actually surprised it worked at all in the VM and was sufficiently impressed when it did. It will only work through VirtualBox though I imagine and I doubt it would work on VirtualBox through Windows.

**Comment by icecube45** [comment]
Was everything made clean after updating aasdk to my fork? (or rsjudk's dev branch)
I've had that issue before where it appeared as it updated and built with the new code, but had something cached that retained the issue.

With the pandemic I haven't really be out driving much, so it's been a hot minute since I've run IA - I'll check it out this weekend when I have some free time and make sure it's not something new.
